tukaan/windows/wm.py

```diff
@@ -557,3 +562,47 @@ def set_backdrop_effect(
 
         self.use_dark_mode_decorations = dark
         self._force_redraw_titlebar()
+
+    @property
+    def window_icon(self) -> Icon | None:
+        return self._window_icon
+
+    @window_icon.setter
+    def window_icon(self, icon: Path | Icon) -> None:
+        if isinstance(icon, Icon):
+            # If the icon was created, then by here Tcl/Tk has to have created a valid photo image for it
+            Tcl.call(None, "wm", "iconphoto", self._wm_path, "-default", icon._name)
+            self._window_icon = icon
+
+        else:
+            icon_path = Path(icon)
+            if icon_path.suffix == ".png":
+                win_icon = Icon(icon_path)
+                self._window_icon = win_icon
+                Tcl.call(None, "wm", "iconphoto", self._wm_path, "-default", win_icon._name)
+            elif icon_path.suffix in (".ico", ".icns"):
+                # Tcl/Tk does not support making photo images or bitmap images directly from .ico or .icns files
+                # But it can make sure that .png files are properly coerced under the hood
+                # So we "cast" the image format before giving it to Tcl/Tk
+                # This casting means it's actually very easy to support a wide array of image file types for icons
+                # in the future; Pillow just needs to be able to read and write to the image type
```

Comment:
> In your opinion do you think it's better to store the Icon or the Path to the internal _window_icon?

If we use `wm iconbitmap` to set an `ico` or `icns` directly, it would make sense to store the `Path` if the file isn't a PNG, and the `Icon` otherwise.

Reply:
It's icns that has multiple resolutions (If you choose to, it can also just have one)

Reply:
I was able to get that working with `wm iconbitmap`. The issue I was running into early is that Tcl cannot instantiate a bitmap using `image create bitmap` from an `.ico` or `.icns` (like one would do for a `photo image`), but passing the file path itself to `wm iconbitmap` works just fine.
Also saved the `_icon` attribute as a `Path` in this case, and tweaked things to only set the internal `_icon` if no errors were thrown.